lib/vagrant-registration/action/register.rb

```diff
@@ -5,34 +5,36 @@ module Registration
     module Action
       # This registers the guest if the guest plugin supports it
       class Register
+        MAX_REGISTRATION_ATTEMPTS = 3
+
         def initialize(app, _)
           @app    = app
           @logger = Log4r::Logger.new('vagrant_registration::action::register')
         end
 
         def call(env)
+          ui = env[:ui]
           # Configuration from Vagrantfile
           config = env[:machine].config.registration
           machine = env[:machine]
           guest = env[:machine].guest
 
-          if should_register?(machine, env[:ui])
-            env[:ui].info I18n.t('registration.action.register.registration_info')
-            check_configuration_options(machine, env[:ui])
+          if should_register?(machine, ui)
+            ui.info I18n.t('registration.action.register.registration_info')
+            check_configuration_options(machine, ui)
 
             unless credentials_provided? machine
               @logger.debug I18n.t('registration.action.register.no_credentials')
 
               # Offer to register ATM or skip
-              register_now = env[:ui].ask I18n.t('registration.action.register.prompt')
+              register_now = ui.ask I18n.t('registration.action.register.prompt')
 
               if register_now == 'n'
                 config.skip = true
               else
-                config = register_on_screen(machine, env[:ui])
+                config = process_registration(guest, machine, ui, config)
```

Comment:
@budhrg method `register_on_screen` was only used here. Now that you have used a new method `process_registration` the old method is an unused code. So why not re-factor `register_on_screen` rather implementing a new method?